x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/search_index/pipelines/ml_inference/add_ml_inference_pipeline_modal.tsx

```diff
@@ -85,6 +85,13 @@ const AddProcessorContent: React.FC<AddMLInferencePipelineModalProps> = ({ onClo
     isLoading,
     addInferencePipelineModal: { step },
   } = useValues(MLInferenceLogic);
+  useEffect(() => {
+    if (createErrors.length === 0) return;
+    const modalOverflow = document.getElementsByClassName('euiModalBody__overflow');
+    if (modalOverflow.length === 0) return;
+    modalOverflow[0].scrollTop = 0;
+  }, [createErrors.join('|')]);
```

Comment:
question(non-blocking): is it better to move this transformation outside of useEffect, or does that not matter for performance?

Reply:
I could move it out. The major thing I was going for here was to not make the `createErrors` array the dependency directly, but use the value instead. Using just `createErrors` caused the hook to be called more than it needed to be.

Reply:
Yeah it's a really elegant solution to that problem. It's fine as-is for me in any case :)